[roultitude] iP

roultitude opened the pull request:
# Donkey Bot

>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

roultitude committed:
Added Greet, Echo, Exit

roultitude committed:
Added functionality for Add, List

roultitude committed:
Added Mark/Unmark functionality, abstracted Tasks into object

roultitude committed:
Added ToDo, Deadline and Event tasks, Fixed String handling for base Task,

roultitude committed:
Added testing for implemented features

Added pause to view runtest results

roultitude committed:
Added error handling

roultitude committed:
Added Delete command

roultitude committed:
Add Saving and Loading to donkey.txt

Add Serialization functionality to Tasks

roultitude committed:
Replace String with LocalDateTime parsing

roultitude committed:
Merge branch 'branch-Level-8'

roultitude committed:
Fix Merge bugs

Type mismatch from String to LocalDateTime in serialization code

roultitude committed:
Abstract task list into TaskList

roultitude committed:
Abstract out saving and serialization to Storage

roultitude committed:
Abstract command processing into Parser

Partial work on abstract user facing interactions to Ui

roultitude committed:
Abstract client facing code to Ui

roultitude committed:
Packaged chatbot into donkeychat

roultitude committed:
Merge branch 'add-gradle-support'

roultitude committed:
Add unit tests for ToDo and TaskList

roultitude committed:
Adjust gradle for jar build

roultitude committed:
Add Javadoc documentation to most public methods and classes

roultitude committed:
Clean up the code to standard

roultitude committed:
Add Find functionality

roultitude committed:
Merge branch 'branch-A-CodingStandard'

roultitude committed:
Merge branch 'branch-A-JavaDoc'

zrei commented on src/main/java/donkeychat/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+
+    /**
+     * Creates a new Deadline task.
+     * @param description Description of the task.
```
Shouldn't there be an empty line between the description and parameter section?

zrei commented on src/main/java/donkeychat/Parser.java:
```diff
@@ -0,0 +1,130 @@
+        Integer taskIndex;
+        try {
+            switch (currCommand) {
+                case "bye":
```
Shouldn't the case clauses have no indentation?

Such as
```java
switch (currCommand) {
case "bye":
    ui.displayBye();
    return false;
```

zrei commented on src/main/java/donkeychat/Task.java:
```diff
@@ -0,0 +1,54 @@
+
+    protected String description;
+    protected boolean isDone;
+    protected final DateTimeFormatter FORMAT_PRINT = DateTimeFormatter.ofPattern("d-MMM-yyyy[ HH:mm]");
```
Shouldn't these `DateTimeFormatter`s be `static`?

zrei commented on src/main/java/donkeychat/Ui.java:
```diff
@@ -0,0 +1,76 @@
+
+public class Ui {
+
+    public void displayIntro() {
```
Shouldn't this public method have a header comment?

zrei commented on src/main/java/donkeychat/Storage.java:
```diff
@@ -0,0 +1,86 @@
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
```
Maybe you can consider separating this import statement from the rest? 

Such as
```java
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;

import java.util.Scanner;
```

zrei reviewed (commented):
You followed the coding standard very well! :+1:

You're just missing an empty line between the description and parameter section for Javadoc comments and incorrect indentation on case clauses in switch statements.

gmengqi commented on src/main/java/donkeychat/Parser.java:
```diff
@@ -0,0 +1,130 @@
+                    ui.displayDeleteTask(taskList.getAtIndex(taskIndex));
+                    taskList.removeAtIndex(taskIndex.intValue());
+                    break;
+                case "find":
```
I think your code is quite easy to follow! Perhaps u might want to align the switch and case to follow the coding standard! You can also consider declaring the commands (find, mark, todo ...) as constants and use it at the case statement as well!

gmengqi commented on src/main/java/donkeychat/Ui.java:
```diff
@@ -0,0 +1,76 @@
+            System.out.println(task);
+        } else {
+            System.out.println("OK, I've marked this task as not done yet:");
+            System.out.println(task);
```
can refactor the code and extract out the System.out.println(task) and put it outside of the if else loop!

gmengqi reviewed (commented):
Overall I think your code looks good, just need to refactor the some of the code to remove duplicate code!

roultitude committed:
Add JavaFX dependency to gradle

roultitude committed:
Add JavaFX UI

Added a graphical user interface to the chatbot.

Moved several initialization method calls from Run() to Start(), and implemented a launcher to call the Start() function.

Having a graphical UI will allow the chatbot to be more personable and should also make the program more user-friendly.

roultitude committed:
Add assertions for key assumptions

Several points in the code should follow some key assumptions during healthy runtime.

Below are assumptions enforced by assertions:

Dependency injections for parser's parse() function should always be non-null objects.

Ui's output string should never be empty string

roultitude committed:
Several Code Quality and Coding Standard cleanups

To keep Parser class readable, extracted logic handling each command into a separate method.

Additionally in accordance with feedback on the PR, adjust javadoc formatting and switch-case formatting to fit coding standard.

roultitude committed:
Merge pull request #2 from roultitude/branch-A-CodeQuality

Several Code Quality and Coding Standard cleanups

roultitude committed:
Merge branch 'master' into branch-A-Assertions

roultitude committed:
Merge pull request #3 from roultitude/branch-A-Assertions

Add assertions for key assumptions

roultitude committed:
Add snooze capability for deadline tasks

roultitude committed:
Change name on GUI to DonkeyChat

roultitude committed:
Add Documentation

Update User Guide
Update user and chatbot profile images
Add UI Sample image

roultitude committed:
Update build.gradle
